Update for docutils 0.19

### Description

Description mostly copied from m2r2's https://github.com/CrossNox/m2r2/pull/49:

`ErrorString` and `SafeString` are deprecated and have been removed in docutils 0.19. We could replace `ErrorString` with `io.error_string` and `SafeString` with Python's builtin `str()` (although this seems breaking for Python 2.x).

See [docutils:HISTORY.txt](https://sourceforge.net/p/docutils/code/HEAD/tree/trunk/docutils/HISTORY.txt) and [docutils:utils/error_reporting.py](https://sourceforge.net/p/docutils/code/HEAD/tree/tags/docutils-0.19/docutils/utils/error_reporting.py) for more information.

### Details

Trying to use sphinx-mdinclude with docutils 0.19, I got:

```
ImportError while loading conftest '/tmp/openapi/tests/conftest.py'.                                                                                                                                     
tests/conftest.py:9: in <module>                                                                                                                                                                         
    from sphinxcontrib.openapi import utils                                                                                                                                                              
.tox/py/lib/python3.10/site-packages/sphinxcontrib/openapi/__init__.py:13: in <module>                                                                                                                   
    from sphinxcontrib.openapi import renderers, directive                                                                                                                                               
.tox/py/lib/python3.10/site-packages/sphinxcontrib/openapi/renderers/__init__.py:5: in <module>                                                                                                          
    from ._httpdomain import HttpdomainRenderer                                                                                                                                                          
.tox/py/lib/python3.10/site-packages/sphinxcontrib/openapi/renderers/_httpdomain.py:11: in <module>                                                                                                      
    import sphinx_mdinclude                                                                                                                                                                              
.tox/py/lib/python3.10/site-packages/sphinx_mdinclude/__init__.py:12: in <module>                                                                                                                        
    from .sphinx import setup                                                                                                                                                                            
.tox/py/lib/python3.10/site-packages/sphinx_mdinclude/sphinx.py:9: in <module>                                                                                                                           
    from docutils.core import ErrorString                                                                                                                                                                
E   ImportError: cannot import name 'ErrorString' from 'docutils.core' (/tmp/openapi/.tox/py/lib/python3.10/site-packages/docutils/core.py)                                                              
ERROR: InvocationError for command /tmp/openapi/.tox/py/bin/python -m pytest tests/ --strict (exited with code 4)
```

* OS: Ubuntu 22.04
* Python version: 3.10.4
* Project version: sphinx_mdinclude==0.5.1
* Can you repro on main?
  Haven't tried, but the import is [still here](https://github.com/jreese/sphinx-mdinclude/blob/f86a78e7db373b46fa53056c1fdb2cf85b93bc66/sphinx_mdinclude/sphinx.py#L9)
* Can you repro in a clean virtualenv?
  Ran with tox, I'm not super familiar with Python's virtual environments but can help reproduce if it were necessary.

--- a/sphinx_mdinclude/sphinx.py
+++ b/sphinx_mdinclude/sphinx.py
@@ -6,10 +6,14 @@ import os
 import os.path
 
 from docutils import io, nodes, statemachine, utils
-from docutils.core import ErrorString
 from docutils.parsers import rst
 from docutils.parsers.rst import directives as rst_directives
-from docutils.utils import SafeString
+
+try:  # new
+    from docutils.utils.error_reporting import ErrorString, SafeString
+except ImportError:  # old
+    from docutils.core import ErrorString  # type: ignore
+    from docutils.utils import SafeString  # type: ignore
 
 from . import RestMarkdown
 

